### PR TITLE
Add association and validation

### PR DESCRIPTION
**Pull request by murajun1978**
### Overview:概要
Modify model for #168 

### Technical changes:技術的変更点
Add association to Ticket and User
Add validation to TicketSubscription

### CODE REVIEW

File: spec/factories/user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :user do
+    email 'test@example.com'
+    password 'password'
+    password_confirmation 'password'
+  end
+end
```
Comment (chimame) [inline spec/factories/user.rb]: EOF😈

File: app/models/ticket_subscription.rb [hunk 2]
```diff
@@ -2,6 +2,15 @@ class TicketSubscription < ApplicationRecord
   belongs_to :ticket
   belongs_to :user
 
+  validates :ticket_id,
+    presence: true
+
+  validates :user_id,
+    presence: true,
+    uniqueness: {
```
Comment (chimame) [inline app/models/ticket_subscription.rb]: I do not oppose adding things.
If you purchase multiple tickets, purchase history may be required separately in preparation for additional purchase.

File: app/models/ticket_subscription.rb [hunk 3]
```diff
@@ -2,6 +2,15 @@ class TicketSubscription < ApplicationRecord
   belongs_to :ticket
   belongs_to :user
 
+  validates :ticket_id,
+    presence: true
```
Comment (chimame) [inline app/models/ticket_subscription.rb]: Do I have only id? I think that this one is better if ticket is necessary.
`validates :ticket, presence: true`

File: app/models/ticket_subscription.rb (hunk 2 above)
Comment (murajun1978) [inline app/models/ticket_subscription.rb]: why need purchase history?

File: app/models/ticket_subscription.rb (hunk 3 above)
Comment (murajun1978) [inline app/models/ticket_subscription.rb]: i'll remove it.

File: app/models/ticket_subscription.rb (hunk 2 above)
Comment (chimame) [inline app/models/ticket_subscription.rb]: I do not need it now.
If you can purchase multiple tickets and you can modify them, you probably need to record your purchase history.
It is supposed to answer purchase inquiries from users.

**Pull request merged**